### PR TITLE
feat: Add `cdktf provider list` command

### PR DESCRIPTION
**Pull request by mutahhir**
Closes: #2229

WIP Images of how it looks like:
image

### CODE REVIEW

File: packages/cdktf-cli/src/lib/dependencies/dependency-manager.ts [hunk 1]
```diff
@@ -384,4 +385,79 @@ export class DependencyManager {
...
+    const regexes = {
+      [Language.GO]: /github.com\/(cdktf|hashicorp)\/cdktf-provider-(.+)-go\//i,
+      [Language.TYPESCRIPT]: /(.+)/i,
+      [Language.CSHARP]: /HashiCorp\.Cdktf\.Provider\.(.+)/i,
+      [Language.JAVA]: /com\.hashicorp\.cdktf-provider-(.+)/i,
+      [Language.PYTHON]: /cdktf-cdktf-provider-(.+)/i,
+    };
+    const regex = regexes[this.targetLanguage];
+    if (!regex) {
+      throw new Error("Language not supported for pre-built providers");
```
Comment (ansgarm) [inline packages/cdktf-cli/src/lib/dependencies/dependency-manager.ts:404]: I think we can use one of our [custom errors](https://github.com/hashicorp/terraform-cdk/blob/86e8afa3d35fdcfeeca73ba6a70872b1fb06a8b1/packages/cdktf-cli/src/lib/errors.ts#L40) here instead

Comment (mutahhir) [reply packages/cdktf-cli/src/lib/dependencies/dependency-manager.ts:404]: Nice! Didn't know about those.

**Comment by mutahhir** [comment]
Rebasing PR to be based on: https://github.com/hashicorp/terraform-cdk/pull/2265 since I need to use the versions.json file from here on.

**Comment by ansgarm** [comment]
> Rebasing PR to be based on: #2265 since I need to use the versions.json file from here on.

It could also conflict with https://github.com/hashicorp/terraform-cdk/pull/2266 (haven't checked, though)

**Pull request closed without merging**